mongodbatlas.go

```diff
@@ -203,8 +229,14 @@ func (m *MongoDBAtlas) Type() (string, error) {
 	return mongoDBAtlasTypeName, nil
 }
 
+// Check to see if the user is a X509 user
+func isX509User(username string) bool {
+	return strings.HasPrefix(username, "CN=")
```

Comment:
This will work. I think a nice improvement for the future would be to have Vault supply the credential type in the `dbplugin.DeleteUserRequest` so that we don't have to write these types of checks. Not blocking but wanted to note this.